src/Data/List/Relation/Unary/Sorted.agda

```diff
@@ -0,0 +1,141 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Inductive pointwise sorting of vectors
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.List.Relation.Unary.Sorted where
+
+open import Data.Nat using (suc; _≤_; _<_; z≤n; s≤s; ⌈_/2⌉; ⌊_/2⌋)
+open import Data.List using (List; []; _∷_; _++_; length)
+open import Data.List.Relation.Unary.All using (All; []; _∷_)
+open import Data.List.Relation.Unary.AllPairs using (AllPairs; []; _∷_)
+open import Data.Maybe as M using (Maybe; nothing; just)
+open import Data.Unit using (⊤; tt)
+open import Data.Product as Σ using (Σ; Σ-syntax; _×_; _,_; uncurry; proj₁; proj₂)
+open import Data.Sum as ⊎ using (_⊎_; inj₁; inj₂)
+open import Relation.Binary as B using (Rel; Total)
+open import Relation.Binary.PropositionalEquality as P using (_≡_)
+open import Relation.Unary as U using (Pred)
+open import Relation.Nullary
+import Relation.Nullary.Decidable as Dec
+open import Relation.Nullary.Product using (_×-dec_)
+open import Induction.WellFounded using (Acc; acc)
+open import Level using (_⊔_; Lift; lift) renaming (suc to lsuc)
+
+Sorted-≺-head : ∀ {a ℓ} {A : Set a} → Rel A ℓ → A → List A → Set ℓ
+Sorted-≺-head _≺_ x [] = Lift _ ⊤
+Sorted-≺-head _≺_ x (y ∷ xs) = x ≺ y
+
+data Sorted {a ℓ} {A : Set a} (_≺_ : Rel A ℓ) : Pred (List A) (a ⊔ ℓ) where
+  [] : Sorted _≺_ []
+  _∷_ : ∀ {x xs} → Sorted-≺-head _≺_ x xs → Sorted _≺_ xs → Sorted _≺_ (x ∷ xs)
+
+Monotonic : ∀ {a ℓ} {A : Set a} → (_≺_ : Rel A ℓ) → Pred (List A) (a ⊔ ℓ)
+Monotonic _≺_ = AllPairs _≺_
```

Comment:
Shouldn't this definition of `Monotonic` go elsewhere? It seems like it could be defined in `AllPairs` for example.